deploy/helm/elastic-agent/templates/agent/k8s/_pod_template.tpl

```diff
@@ -0,0 +1,133 @@
+{{- define "elasticagent.engine.k8s.podTemplate" }}
```

Comment:
Does this need to be a template function? Could we not make a *.yaml file for this and include that in the manifests? It'd be nicer to work with due to syntax highlighting and such.

Reply:
how do you include dynamically at will *.yaml templates @swiatekm ? because even [here](https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-collector/templates/_pod.tpl) the equivalent pod template needs to be a *.tpl to be included dynamically

Reply:
Here's an example, the otel-operator Chart has a templated CRD in the form of a yaml file: https://github.com/open-telemetry/opentelemetry-helm-charts/blob/main/charts/opentelemetry-operator/conf/crds/crd-opentelemetrycollector.yaml
and this is included dynamically via a combination of `tpl` and `Files.Get`: https://github.com/open-telemetry/opentelemetry-helm-charts/blob/b8d11b63916bc40cd1aa2875a880f3995978d434/charts/opentelemetry-operator/templates/admission-webhooks/operator-webhook.yaml#L273.

Reply:
@swiatekm this example essentially loads up the contents of the yaml file and then evaluates them as template which is practically the same as having directly a *.tpl. With the former approach though you have now to maintain also file structures where with the tpl approach you can reference the template from anywhere by name. What exactly do we gain by switching to this paradigm?

Reply:
The benefit is that a *.yaml file can get YAML syntax highlighting from most tools, and often even K8s manifest linting. It's usually also obvious where the templating happens. In comparison tpl files get none of that, and non-tpl text in them is just treated as plain text.

In this very PR, reviewing the templated K8s manifest files is much easier than reviewing the tpl files, for this simple reason.

Reply:
but @swiatekm the tpl under debate essentially fills up the spec of the podTemplate of either a `daemonset`, `statefulset` or `deployment`. Even if I switched this to *.yaml, how you would get linting since this yaml will still not feature any of the required yaml fields, such `apiVersion:`, `kind: ...`, etc., so your editor can give you proper linting?

Reply:
I think what you are after is essentially duplicating this template in all daemonset, statefulset or deployment and having each k8s object as a whole. However there is an upper limit on the size of the releases and all these templates when you do a helm install are compressed and saved inside a kubernetes configmap. Thus since this pod spec is the same for all three deployment options it became a single tpl file

Reply:
I don't actually want to duplicate this specific template, I'd just like to make it easier to read. This matters less for smaller templates, like the volumes or env variables, but for the pod template I think it's worth it to make it a separate yaml file.

If you're really against it, I won't push, but I do think doing this would make things easier for us in the long term.

Reply:
@swiatekm I think you focused just on my latest reply, but what about the missing yaml fields required for linting that I mention [here](https://github.com/elastic/elastic-agent/pull/5331#discussion_r1740979067)? do you believe that by making this yaml without having the aforementioned fields will "help" your IDE/text editor to provide linting about pod spec?

Reply:
not sure what IDE you are using but for my case - goland  + kubernetes plugin - YAML (not k8s specific object) highlighting works for templates. To this end, even when I did the change to yaml locally I couldn't find an option to enforce this yaml to be parsed as the spec of a pod template, and hence get k8s specific linting. For now I will resolve this but if you have more hassle-free steps that we could use to get former, for sure, we can switch to that

Reply:
That's all right. After we get this merged, I'll put up a draft PR to show you what change I had in mind exactly.